### ISSUE TITLE
Running the plan with google_compute_instance_template results in an error

### ISSUE THREAD

**Issue by HirokiSakonju** [terraform-providers/terraform-provider-google#2067]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version

```
$ terraform -v
Terraform v0.11.8
+ provider.google v1.18.0
```

### Affected Resource(s)

google_compute_instance_template

### Terraform Configuration Files

```tf
resource "google_compute_instance_template" "instance-01" {
  name           = "instance-01"
  description    = "instance-01"
  can_ip_forward = true
  machine_type   = "n1-standard-1"

  scheduling {
    automatic_restart   = true
    on_host_maintenance = "MIGRATE"
  }

  // Create a new boot disk from an image
  disk {
    source_image = "instance01"
    auto_delete  = true
    boot         = true
    disk_size_gb = 10
  }

  network_interface {
    subnetwork = "${google_compute_subnetwork.intra-network.name}"

    access_config {
      nat_ip = "${google_compute_address.instance-01.address}"
    }
  }

  service_account {
    scopes = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]
  }
}
```

### Expected Behavior

I did not change it.

### Actual Behavior

`terraform plan`

```
* google_compute_instance_template.instance-01: 1 error(s) occurred:

* google_compute_instance_template.instance-01: google_compute_instance_template.instance-01: error flattening disks: Error getting relative path for source image: String was not a self link: global/images/instance01
```

### Important Factoids

The image specified by source_image is an image of an image created by another project.
It is not an image existing by default in GCP.

Also, this google_compute_instance_template is a resource you have already created.

### References

This correction seems to have caused this problem.

* #1916

**Comment by marrau** [comment]
This also applies to my configurations. I tried to get a self_link using data.google_compute_image (which actually resolves it correctly), but i´m still getting the error as mentioned.

**Comment by caarlos0** [comment]
also happens with:

```terraform
resource "google_compute_instance_template" "foo_1" {
  machine_type   = "n1-standard-2"
  region         = "${var.gcp_region}"
  tags           = ["${var.gcp_network_private_tag}", "fooo"]
  can_ip_forward = false
  disk {
    source_image = "ubuntu-os-cloud/ubuntu-1404-lts"
    disk_size_gb = "60"
    auto_delete  = true
    boot         = true
  }
  disk {
    auto_delete  = true
    boot         = false
    disk_size_gb = "100"
    disk_type    = "pd-ssd"
  }
  metadata {
    block-project-ssh-keys = false
  }
  network_interface {
    subnetwork = "${google_compute_subnetwork.subnet_private.name}"
  }
  lifecycle {
    create_before_destroy = true
  }
}
```

**Comment by stanzheng** [comment]
I ran into this issue and found a resolution. Two prong approach

1) referencing the source_image using a source link using https://www.terraform.io/docs/providers/google/d/datasource_compute_image.html

2) running a `terraform apply -refresh=false ...` 

These two combinations allowed me to resolve my initial error of flattening disks and allowed my `tf.state` to resolve.

I think I got into this erroneous state by being allowed to create my instance templates using `string` references to images but requiring refresh of state subsequently to have a `self_link` 

Fix: disallow source_image to take a string?
